crates/bevy_ecs/src/schedule/schedule.rs

```diff
@@ -549,6 +549,73 @@ impl Schedule {
             self.executor.run(
                 &mut self.executable,
                 world,
+                None,
+                skip_systems.as_ref(),
+                error_handler,
+            );
+        }
+    }
+
+    /// Runs all systems in the specified system set (including transitively) on
+    /// the `world`, using its current execution strategy.
+    pub fn run_system_set<M>(&mut self, world: &mut World, set: impl IntoSystemSet<M>) {
```

Comment:
Do you think there would be value to combine the internals of this method with the `run` method to avoid code duplication and make this more maintainable?

Maybe a `run_internal(&mut self, world: &mut World, set: Option<impl IntoSystemSet<M>)` method?